man/fi_cxi.7.md

```diff
@@ -0,0 +1,1782 @@
+message rate outweighs by the cost of restarting after a failure.
+
+Unreliable, non-fetching AMOs may be performed by specifying the
+*FI_CXI_UNRELIABLE* flag. Unreliable, fetching AMOs are not supported. Unreliable
+AMOs must target an optimized MR and cannot use remote completion notification.
+Unreliable AMOs are not ordered.
+
+## High Rate Put
+
+High Rate Put (HRP) is a feature that increases message rate performance of RMA
+and unreliable non-fetching AMO operations at the expense of global ordering
+guarantees.
+
+HRP responses are generated by the fabric egress port. Responses are coalesced
+by the fabric to achieve higher message rates. The completion event for an HRP
+operation guarantees delivery but does not guarantee global ordering. If global
+ordering is needed following an HRP operation, the source may follow the
+operation with a normal, fenced Put.
+
+HRP RMA and unreliable AMO operations may be performed by specifying the
+*FI_CXI_HRP* flag. HRP AMOs must also use the *FI_CXI_UNRELIABLE* flag. Monitor the
+hardware counter C_CNTR_HNI_HRP_ACK at the initiator to validate that HRP is in
+use.
+
+## Counters
+
+Cassini offloads light-weight counting events for certain types of operations.
+The rules for offloading are:
+
+* Counting events for RMA and AMO source events are always offloaded.
+* Counting events for RMA and AMO target events are always offloaded.
+* Counting events for Sends are offloaded when message size is less than the
+  rendezvous threshold.
+* Counting events for message Receives are never offloaded by default.
+
+Software progress is required to update counters unless the criteria for
+offloading are met.
+
+# RUNTIME PARAMETERS
+
+The CXI provider checks for the following environment variables:
+
+*FI_CXI_ODP*
+:   Enables on-demand paging. If disabled, all DMA buffers are pinned.
+    If enabled and mr_mode bits in the hints exclude FI_MR_ALLOCATED,
+    then ODP mode will be used.
+
+*FI_CXI_FORCE_ODP*
+:   Experimental value that can be used to force the use of ODP mode
+    even if FI_MR_ALLOCATED is set in the mr_mode hint bits. This is
+    intended to be used primarily for testing.
+
+*FI_CXI_ATS*
+:   Enables PCIe ATS. If disabled, the NTA mechanism is used.
+
+*FI_CXI_ATS_MLOCK_MODE*
+:   Sets ATS mlock mode. The mlock() system call may be used in conjunction
+    with ATS to help avoid network page faults. Valid values are "off" and
+    "all". When mlock mode is "off", the provider does not use mlock(). An
+    application using ATS without mlock() may experience network page faults,
+    reducing network performance. When ats_mlock_mode is set to "all", the
+    provider uses mlockall() during initialization with ATS. mlockall() causes
+    all mapped addresses to be locked in RAM at all times. This helps to avoid
+    most network page faults. Using mlockall() may increase pressure on
+    physical memory.  Ignored when ODP is disabled.
+
+*FI_CXI_RDZV_THRESHOLD*
```

Comment:
I understand how it would be nice to use consistent naming, but this provider is already being used and many environments are being setup using this naming. We could always map both to the same value, but I personally think that creates more confusion from a users perspective.

Reply:
I would like to consider this as a future refinement. I understand where you are coming from @j-xiong , but we will need time to transition users to a new environment variable and deprecate this one over a period of time.